numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1738,12 +1738,21 @@ unpack_bits(PyObject *input, int axis)
     }
 
     /* Setup output shape */
-    for (i=0; i<PyArray_NDIM(new); i++) {
+    for (i = 0; i < PyArray_NDIM(new); i++) {
         outdims[i] = PyArray_DIM(new, i);
     }
 
     /* Multiply axis dimension by 8 */
-    outdims[axis] <<= 3;
+    if (count <= 0) {
+        count += (outdims[axis] << 3);
```

Comment:
`* 8` would be much clearer here

Reply:
Strangely enough, I decided to keep the existing convention for this bit. I rather prefer `<< 3` for some reason.